drivers/interrupt_controller/intc_ioapic.c

```diff
@@ -83,7 +83,7 @@ DEVICE_MMIO_TOPLEVEL_STATIC(ioapic_regs, DT_DRV_INST(0));
  * In either case, regardless how many CPUs in the system, 0xff implies that
  * it's intended to deliver to all possible 8 local APICs.
  */
-#define DEFAULT_RTE_DEST	(0xFF << 24)
+#define DEFAULT_RTE_DEST	0xFF000000U
```

Comment:
please do not do this (here and in al the other places). Having the value splitted in mask and offset makes the code more readable because we can identify the bit field length and offset at first sight.

Reply:
Unfortunately things are a little more complex than that: according to MISRA (and for portability sake) the left shift should be applied to a constant that has provably a wideness compatible with the shift count on any arch, here suggesting something like:
#define DEFAULT_RTE_DEST ((uint32_t)0xFF << 24)
but if this is macro is used in a preprocessor condition this cannot be done.
This is the only case where I have been forced to do that.
If you agree I'll change it in:

#define DEFAULT_RTE_DEST	0xFF000000U /* 0xFF < 24 */

Reply:
> #define DEFAULT_RTE_DEST 0xFF000000U /* 0xFF << 24 */

I'd say it makes situation even worse - as we may have different values in code and in comments.
And it also makes code less readable.

I'd prefer to keep original variant.

Reply:
I don't see how this makes code less readable.
Also take in account that to have an implementation and a comment clarifying the specification is a common pattern. I'd say it is the very purpose of code comments.
Consider also that personal preferences are not always secondable, please don't expect from me so much ;-)

Reply:
This definitely makes the code less readable and more redundant and this is not matter of personal preference.

Especially in the context of embedded systems it is extremely common to have MASK and OFFSET that are directly derived from the datasheets as defines so that you can do:
```
reg |= ((val & MASK) << OFFSET);
```
Being forced to define MASK, OFFSET (to be used in the code) **and** the extended value for all the other cases makes no sense.

So, I also prefer to keep the original variant.